src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -0,0 +1,257 @@
+            this.Values = values;
+            this.EndTime = default;
+            this.LongValue = default;
+            this.longVal = default;
+            this.lastLongSum = default;
+            this.DoubleValue = default;
+            this.doubleVal = default;
+            this.lastDoubleSum = default;
+
+            if (this.AggType == AggregationType.Histogram)
+            {
+                this.ExplicitBounds = new double[] { 0, 5, 10, 25, 50, 75, 100, 250, 500, 1000 };
+                this.BucketCounts = new long[this.ExplicitBounds.Length + 1];
+                this.bucketCounts = new long[this.ExplicitBounds.Length + 1];
+                this.lockObject = new object();
+            }
+            else
+            {
+                this.ExplicitBounds = null;
+                this.BucketCounts = null;
+                this.bucketCounts = null;
+                this.lockObject = null;
+            }
+        }
+
+        public string[] Keys { get; internal set; }
+
+        public object[] Values { get; internal set; }
+
+        public DateTimeOffset StartTime { get; internal set; }
+
+        public DateTimeOffset EndTime { get; internal set; }
+
+        public long LongValue { get; internal set; }
+
+        public double DoubleValue { get; internal set; }
+
+        public long[] BucketCounts { get; internal set; }
+
+        public double[] ExplicitBounds { get; internal set; }
+
+        private readonly AggregationType AggType { get; }
+
+        internal void Update(long number)
+        {
+            switch (this.AggType)
+            {
+                case AggregationType.LongSumIncomingDelta:
+                    {
+                        Interlocked.Add(ref this.longVal, number);
+                        break;
+                    }
+
+                case AggregationType.LongSumIncomingCumulative:
+                    {
+                        Interlocked.Exchange(ref this.longVal, number);
+                        break;
+                    }
+
+                case AggregationType.LongGauge:
+                    {
+                        Interlocked.Exchange(ref this.longVal, number);
+                        break;
+                    }
+
+                case AggregationType.Histogram:
+                    {
+                        this.Update((double)number);
+                        break;
+                    }
+            }
+        }
+
+        internal void Update(double number)
+        {
+            switch (this.AggType)
+            {
+                case AggregationType.DoubleSumIncomingDelta:
+                    {
+                        double initValue, newValue;
+                        do
+                        {
+                            initValue = this.doubleVal;
+                            newValue = initValue + number;
+                        }
+                        while (initValue != Interlocked.CompareExchange(ref this.doubleVal, newValue, initValue));
```

Comment:
Thanks! 😄 I learned something here... `Interlocked.Add` equivalent for `double`

Reply:
it has scope for improvement. I think we need to put a hard limit like this:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Internal/CircularBuffer.cs#L149-L152